### PR TITLE
Add Amazon Services transport truststore configuration

### PR DESCRIPTION
**Pull request by antonwiens**
Implementation for #10888.

I still need to test things out.

But could you already look at the design of the implementation, if its ok for you or if you have suggestions to implement this in a better way.

### CODE REVIEW

**Comment by antonwiens** [comment]
I am waiting for #11374 so i can test it out in my project.

**Comment by gsmet** [comment]
@marcinczeczko could you have a look at that one if you're back from vacation? :)

**Comment by marcinczeczko** [comment]
@gsmet thanks for the remainder. Of course I'm back month ago but simply forgot :) will take a look tomorrow, but my first spot are the missing tests. Dynamodb/runtime module has lot of tests

File: extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/FileTlsTrustManagersProvider.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import software.amazon.awssdk.http.TlsTrustManagersProvider;
+
+public class FileTlsTrustManagersProvider implements TlsTrustManagersProvider {
```
Comment (marcinczeczko) [inline extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/FileTlsTrustManagersProvider.java:20]: Please add javadoc here explaining the purpose of this class

File: extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/FileTlsTrustManagersProvider.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        type = fileStore.type.get();
+        password = fileStore.password.map(String::toCharArray).orElse(null);
+    }
+
+    @Override
+    public TrustManager[] trustManagers() {
+        try (InputStream storeInputStream = Files.newInputStream(path)) {
+            KeyStore keyStore = KeyStore.getInstance(type);
+            keyStore.load(storeInputStream, password);
+            TrustManagerFactory tmf = TrustManagerFactory.getInstance("SunX509");
```
Comment (marcinczeczko) [inline extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/FileTlsTrustManagersProvider.java:37]: Maybe it's safer to use platform default trustmanager algorithm via `TrustManagerFactory.getDefaultAlgorithm()` ?

File: extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/NoneTlsTrustManagersProvider.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package io.quarkus.amazon.common.runtime;
+
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import software.amazon.awssdk.http.TlsTrustManagersProvider;
+
+public class NoneTlsTrustManagersProvider implements TlsTrustManagersProvider {
```
Comment (marcinczeczko) [inline extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/NoneTlsTrustManagersProvider.java:10]: Same here, please add some javadoc here

File: extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/SystemPropertyTlsTrustManagersProvider.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package io.quarkus.amazon.common.runtime;
+
+import javax.net.ssl.TrustManager;
+
+import software.amazon.awssdk.http.TlsTrustManagersProvider;
+
+public class SystemPropertyTlsTrustManagersProvider implements TlsTrustManagersProvider {
```
Comment (marcinczeczko) [inline extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/SystemPropertyTlsTrustManagersProvider.java:7]: couple words explanations  (javadoc) would be usefull

File: extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/TlsTrustManagersProviderConfig.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+@ConfigGroup
+public class TlsTrustManagersProviderConfig {
+
+    // @formatter:off
+    /**
+     * TLS trust managers provider type.
+     *
+     * Available providers:
+     *
+     * * `insecure` - Use this provider if you don't want to validate the servers certificates.
```
Comment (marcinczeczko) [inline extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/TlsTrustManagersProviderConfig.java:15]: maybe `trust-all` would be a better name, wdyt ?

**Review by marcinczeczko — CHANGES_REQUESTED** [review]
@antonwiens it looks great. I posted just couple of cosmetic comments. Besides that please update tests in the `amazon-services/dynamodb/deployment` module as they are two (sync/async) verifying if config with defined keystore loads correctly

File: extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/FileTlsTrustManagersProvider.java (hunk 1 above)
Comment (antonwiens) [reply extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/FileTlsTrustManagersProvider.java:20]: Will do it.

File: extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/TlsTrustManagersProviderConfig.java (hunk 5 above)
Comment (antonwiens) [reply extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/TlsTrustManagersProviderConfig.java:15]: Trusting all certificates is generally a bad thing to do, thats why i choose a "negative" name. But i can change it to "trust-all" if you prefer it.

**Comment by antonwiens** [comment]
@marcinczeczko I would like to add a test for 
quarkus.dynamodb.async-client.tls-trust-managers-provider.type=file-store
Is this possible? what would be the filepath, etc., for the filestore?

**Comment by antonwiens** [comment]
> @marcinczeczko I would like to add a test for
> `quarkus.dynamodb.async-client.tls-trust-managers-provider.type=file-store`
> Is this possible? what would be the filepath, etc., for the filestore?

I added some tests, can you check them please.

**Comment by antonwiens** [comment]
I just successfully completed the integrationtest in my private project with an s3 service and the truststore configuration works for it in jvm and in native mode

**Comment by marcinczeczko** [comment]
@antonwiens - I just reviewed and it looks good to me. Thanks !

**Comment by gsmet** [comment]
I squashed everything, rebased and force-pushed.

File: extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/TlsTrustManagersProviderConfig.java (hunk 5 above)
Comment (gsmet) [reply extensions/amazon-services/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/TlsTrustManagersProviderConfig.java:15]: I'm a bit late to the party but let's go with `trust-all`. It's more consistent with the other parts of the code.

**Review by gsmet — CHANGES_REQUESTED** [review]

**Comment by antonwiens** [comment]
Seems like an unrelated test is failing?

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**